packages/clerk-js/src/ui/UserProfile/MfaBackupCodeCreatePage.tsx

```diff
@@ -0,0 +1,181 @@
+      .catch(err => handleError(err, [], card.setError));
+  }, []);
+
+  if (card.error) {
+    return <ContentPage.Root headerTitle={title} />;
+  }
+
+  if (!backupCode) {
+    return (
+      <Spinner
+        colorScheme='primary'
+        size='lg'
+      />
+    );
+  }
+
+  return (
+    <ContentPage.Root headerTitle={title}>
+      <Text
+        localizationKey={text}
+        variant='regularRegular'
+      />
+      <MfaBackupCode backupCodes={backupCode.codes} />
+
+      <FormButtonContainer>
+        <NavigateToFlowStartButton
+          variant='solid'
+          autoFocus
+          localizationKey={localizationKeys('userProfile.formButtonPrimary__finish')}
+        />
+      </FormButtonContainer>
+    </ContentPage.Root>
+  );
+};
+
+type MfaBackupCodeProps = {
+  backupCodes?: string[];
+};
+
+const MfaBackupCode = (props: MfaBackupCodeProps) => {
+  const { backupCodes } = props;
+  const { applicationName } = useEnvironment().displayConfig;
+  const user = useCoreUser();
+  const { print, printableProps } = usePrintable();
+  const { onCopy, hasCopied } = useClipboard(backupCodes?.join(',') as string);
+
+  const onDownloadTxtFile = () => {
+    const element = document.createElement('a');
+    const file = new Blob([txtFileContent(backupCodes, applicationName, getIdentifier(user))], {
+      type: 'text/plain',
+    });
+    element.href = URL.createObjectURL(file);
+    element.download = `${applicationName}_backup_codes.txt`;
+    document.body.appendChild(element);
+    element.click();
+  };
+
+  if (!backupCodes) {
+    return null;
+  }
+
+  return (
+    <>
+      <Col gap={1}>
+        <Text
+          localizationKey={'Backup codes'}
+          variant='regularMedium'
+        />
+        <Text
+          localizationKey={'Store them securely and keep them secret.'}
+          variant='smallRegular'
+          colorScheme='neutral'
+        />
+      </Col>
+      <Grid
+        gap={2}
+        sx={t => ({
+          gridTemplateColumns: `repeat(5, minmax(${t.sizes.$12}, 1fr))`,
+          [mqu.sm]: {
+            gridTemplateColumns: `repeat(2, minmax(${t.sizes.$12}, 1fr))`,
+          },
+        })}
+      >
+        {backupCodes.map((code, i) => (
+          <BackupCodeTile
+            key={i}
+            code={code}
+          />
+        ))}
+      </Grid>
+
+      <Flex gap={6}>
+        <Button
+          variant='link'
+          colorScheme='primary'
+          onClick={onCopy}
+        >
+          {!hasCopied ? 'Copy all' : 'Copied!'}
+        </Button>
+
+        <Button
+          variant='link'
+          colorScheme='primary'
+          onClick={onDownloadTxtFile}
+        >
+          Download .txt
+        </Button>
+
+        <Button
+          variant='link'
+          colorScheme='primary'
+          onClick={print}
+        >
+          Print
+        </Button>
+      </Flex>
+
+      <PrintableComponent {...printableProps}>
+        <Heading>Your backup codes are:</Heading>
```

Comment:
🙃 Should we also include the  application name and possibly an identifier?

Reply:
Added both in the same way we do for the txt file